sort/src/main.rs

```diff
@@ -36,41 +29,31 @@ fn main_sort(matches: clap::ArgMatches) -> Result<(), SortError> {
 }
 
 fn get_inputs(matches: &clap::ArgMatches) -> Result<Vec<Box<dyn Read>>, SortError> {
-    use std::io::Error;
-
-    match matches.values_of("FILE") {
-        Some(m) => {
-            println!("{:?}", m);
-            let files: Vec<Result<File, Error>> = m.map(File::open).collect();
-            let files: Result<Vec<File>, Error> = files.into_iter().collect();
-
-            match files {
-                Ok(files) => {
-                    let mut inputs: Vec<Box<dyn Read>> = vec![];
-                    for file in files {
-                        inputs.push(Box::new(file));
-                    }
-                    Ok(inputs)
-                },
-                Err(err) => Err(SortError::FileReadError(err)),
+    match matches.values_of("INPUT_FILES") {
+        Some(files) => {
+            let files: io::Result<Vec<File>> = files.map(File::open).collect();
+            let files = files.map_err(SortError::FileReadError)?;
+
+            let mut inputs: Vec<Box<dyn Read>> = vec![];
+            for file in files {
+                inputs.push(Box::new(file));
             }
+            Ok(inputs)
         },
-        None => Ok(vec![Box::new(stdin())]),
+        None => Ok(vec![Box::new(io::stdin())]),
     }
 }
 
-fn sort(
-    flags: &SortFlags, inputs: Box<dyn Iterator<Item = Box<dyn Read>>>,
```

Comment:
I think that there was no need to make this a Box of Boxes, so I simplified it into a `Vec`, because the generics were already transforming it into a `Vec`.